interchain/tests/tests/migrate.rs

```diff
@@ -0,0 +1,57 @@
+mod common;
+
+use abstract_core::objects::AccountId;
+use abstract_interface::Abstract;
+use abstract_interface::*;
+use anyhow::Ok;
+use cosmwasm_std::Addr;
+use cw20::Cw20QueryMsg;
+use cw_orch::daemon::networks::JUNO_1;
+use cw_orch::prelude::*;
+use cw_orch_fork_mock::ForkMock;
+use std::path::PathBuf;
+use std::str::FromStr;
+use tokio::runtime::Runtime;
+
+use cosmwasm_std::Empty;
+
+const VERSION: &str = env!("CARGO_PKG_VERSION");
+// owner of the abstract infra
+const SENDER: &str = "juno1kjzpqv393k4g064xh04j4hwy5d0s03wfvqejga";
+
+/// Sets up the forkmock for Juno mainnet.
+/// Returns the abstract deployment and sender (=mainnet admin)
+fn setup() -> anyhow::Result<(Abstract<ForkMock>, Addr, ForkMock)> {
+    let runtime = Runtime::new().unwrap();
+    env_logger::init();
+    let sender = Addr::unchecked(SENDER);
+    // Run migration tests against Juno mainnet
+    let mut app = ForkMock::new(&runtime, JUNO_1)?;
+    app.set_sender(sender.clone());
+
+    let abstr_deployment = Abstract::load_from(app.clone())?;
+    Ok((abstr_deployment, sender, app))
+}
+
+#[test]
+fn migrate_infra_success() -> anyhow::Result<()> {
+    let (abstr_deployment, sender, _) = setup()?;
+
+    assert_eq!(abstr_deployment.version_control.code_id()?, 3692);
+    abstr_deployment.migrate_if_needed()?;
+    assert_eq!(abstr_deployment.version_control.code_id()?, 5000003);
```

Comment:
don't think code_ids should be consts here

Reply:
Then where should I retrieve them? They won't change (unless we change the network we test against) 

We'll encounter the same issue for tests related to balances...

Reply:
Ohhh but I see what you're saying. this code-id will change when we do our next migration...

Reply:
> Ohhh but I see what you're saying. this code-id will change when we do our next migration...

It will change on the next commit actually :rofl:
Uh I just realized that we can't check previous version code_id from Abstract object, right? I guess we can parse it from github/last_version_tag/state.json, but that could be easier to do manually

Reply:
> It will change on the next commit actually 🤣

Nvm I didn't know `ForkMock` starts with hardcoded code_id

Reply:
Yup! It's because cw-multi-test does that as well afaik